guardrails/validators.py

```diff
@@ -613,6 +613,14 @@ def validate(self, key: str, value: Any, schema: Union[Dict, List]) -> Dict:
             f"Validating {value} is in length range {self._min} - {self._max}..."
         )
 
+        # TODO: should type hint of value be Any, if we rely on calling len()
+        # on it?
+
+        # Make value explicitly a string now. Optionally validation could be
+        # directly failed, but there could be cases where nested dictionaries
+        # are desired..
+        value = str(value)
```

Comment:
The `valid-length` validator should also support lists, so instead of the string casting here another option could be to add an assert that the `value` is of type string or list.

Reply:
I see, thanks! What behavior is expected from a string like {"value": "something"} that gets passed in as a dict? Should str conversion only be done if value is not a list, or should the validation fail if value is not list or str?

Reply:
Ah interesting, in this case the validation should fail if value is not a list or a string!

Reply:
Just making sure this should happen inside the `validate()` function, or is there (should there be?) a mechanism that checks the allowed data types prior to `validate()`, given that there is also the data type entered to `@register_validator`?

Reply:
@Mikkolehtimaki ideally this should be tested in the data type's `from_str` method. Then, in the data type's `validate` method, there should be a deterministic fix or a ReAsk object that is created in case the value doesn't match the data type's value. This would require changing some of the dependencies between `datatypes.py` and `validators.py`.

For now, it makes sense to address this in a separate PR, and simply failing validation in case the value isn't a string or a dict. It shouldn't be too difficult to change the dependencies -- I can take a stab at it!